inception/inception-search-core/src/main/java/de/tudarmstadt/ukp/inception/search/SearchService.java

```diff
@@ -108,6 +109,18 @@ long determineNumOfQueryResults(User aUser, Project aProject, String aQuery,
             SourceDocument aDocument, AnnotationLayer aAnnotationLayer,
             AnnotationFeature aAnnotationFeature)
         throws ExecutionException, IOException;
+    /*
+    public HashMap<String, HashMap<String, Double>> getProjectTextStatistics(User aUser,
+            Project aProject, String aStatistic, SourceDocument aDocument,
+            AnnotationLayer aAnnotationLayer, AnnotationFeature aAnnotationFeature,
+            String[] aPunctuationMarks)
+        throws IOException, ExecutionException;
+
+     */
+
+    public Map<String, Map<String, Object>> getProjectStatistics(User aUser, Project aProject, String aStatistic,
+             Double aLowerDocSize, Double aUpperDocSize)
```

Comment:
Why are these Double and not double?

Reply:
By accident. Changed it to double

Reply:
Well, just realised that it is actually needed. To also consider unbounded documents the mtas method requires null for upperDocSize. Thus we need Double and not double.

Reply:
It seems quite curious that a document size can be a float number - any idea what unit the document size is measured in? There is also OptionalDouble that could be considered here.

Reply:
I'm pretty sure that the document number is the number of tokens of a document :
![grafik](https://user-images.githubusercontent.com/61779598/132341803-0e9a3c82-eb85-4420-9cd4-d155b959f8de.png)
( taken from the mtas documentation https://textexploration.github.io/mtas/search_component_stats_tokens.html )

I need this method which wants a Double; a Null indicates unboundedness in this direction.
![grafik](https://user-images.githubusercontent.com/61779598/132341648-42719ae3-f83f-46d5-b3ba-2a5a00a0ddb8.png)

Reply:
You do no necessarily need to use a Double on our own method signatures. E.g. you could use an `OptionalInt aMinTokens` and instead of passing this value directly to the MTAS method, you could do `Double minTokens = aMinTokens.isPresent() ? (double) aMinTokens.get() : null`

Reply:
I can do it this way if you want. But what is the additional value of using OptionalInt over Double? For me, it seems to be more complicated.

Reply:
It is. But the `Double` raises many questions such as "what is the unit?" and "why can it be null?". If the unit is a token, then it raises the question "why can there be fractions of a token?". An `OptionalInt aMaxTokens`would not raise these questions, because the unit is clear, it is clear that the limit is optional, and there is no question about fractions of tokens.